### ISSUE TITLE
Tooltip doesn't show when button is disabled

### ISSUE THREAD

**Issue by MikeyZat** [carbon-design-system/carbon#8509]
# When button is in disabled state, the tooltip doesn't show anymore

## What package(s) are you using?

- [x] `carbon-components: 10.29.0`
- [x] `carbon-components-react: 7.29.0`

## Detailed description

Issue is visible in storybook: https://react.carbondesignsystem.com/?path=/story/components-button--playground

![image](https://user-images.githubusercontent.com/41756225/116234449-e83be000-a75c-11eb-8604-a66e350aeb1b.png)

I would expect the tooltip to show even if button is disabled (this was working before). We often use that option to indicate why the button is disabled.

## Steps to reproduce the issue

1. Create iconOnly button
2. set it to disabled state (`disable` props)
3. See that the tooltip with description doesn't show anymore

## Additional information

This is a regression, it was working before

**Comment by tw15egan** [comment]
@MikeyZat Do you know which version we had disabled tooltips on Icon Only buttons? I've created a CodeSandbox going back to `7.6.0` which is almost 2 years old and I'm not seeing any tooltips when the Button is disabled

https://codesandbox.io/s/happy-cloud-j7n6k?file=/src/index.js

Tooltips on disabled buttons are inherently inaccessible, as keyboard and screen reader users will not be able to focus them, leaving the content and information hidden.

**Comment by MikeyZat** [comment]
I am not able to find correct version which shows tooltips on disabled buttons but I'm almost sure it worked on some.

Anyway, are there any plans to add that functionality? It could be another prop like `disabledButtonDescription`. It would be very helpful in cases we want to inform the user why he can't submit some action.

I can make a PR if given a green light for that.

**Comment by tw15egan** [comment]
@MikeyZat there are no current plans for adding support for tooltips on disabled Buttons. @dakahn, is there any possible workaround for this functionality, or is this pattern inherently inaccessible?

**Comment by leximarie** [comment]
@tay1orjones @dakahn Can we re-open this issue? There is definitely a use case for this. I want a tooltip on an edit icon button to tell the user why they can't edit. This could be a separate prop. 

<img width="372" alt="image" src="https://user-images.githubusercontent.com/11201028/226047742-5f6f37b6-6dfc-4038-9399-12973e3cec39.png">

**Comment by jomaresch** [comment]
I agree with @leximarie. We face the same problem.
In my option this type of user feedback is very elegant and it's even used by Github:
<img width="657" alt="Bildschirmfoto 2023-04-06 um 13 58 13" src="https://user-images.githubusercontent.com/18600999/230374310-7780a819-19ee-45dc-ac48-6471c36e5e35.png">

> That said I would never advise a tooltip be the lone method of feedback for a disabled button.

@dakahn What is the intended way to indicate to the user why a button is disabled?

**Comment by tay1orjones** [comment]
@leximarie Yes, I think it makes sense to reopen. Here's an overview of the status for this:

The core of this issue is that buttons marked `disabled` aren't focusable. In order for tooltip contents to be surfaced to keyboard and screenreader users, tooltips must only be associated with interactive/focusable elements. This is how Carbon Popover, Tooltip, ToggleTip, etc work.

In the example from GitHub, the "approve" button is `disabled` and the tooltip only appears on hover. This is problematic because a keyboard user will never see that message. The tooltip contents is also not read by screenreaders for the same reason.

We're exploring the feasibility of allowing focus on buttons that are functionally disabled in #14646. I'm not confident that it's an accessible pattern though. We're reviewing it as a team, with IBM's Accessibility team, and internally ([slack thread](https://ibm-studios.slack.com/archives/C0M053VPT/p1694528780737269)) with product teams.

All this said, the most valid thing that must be solved is what @jomaresch already outlined:

> What is the intended way to indicate to the user why a button is disabled?

We do not have a good solution for this right now. It's highly dependent on context of the type of button, placement in the overall UI, and the button's function. If we can't land #14646, I think we should identify and document alternative ways to describe disabled elements.

**Comment by tay1orjones** [comment]
> In the example from GitHub, the "approve" button is disabled and the tooltip only appears on hover. This is problematic because a keyboard user will never see that message. The tooltip contents is also not read by screenreaders for the same reason.

GitHub recently updated this and no longer has the tooltip on hover, additionally the text previously in the tooltip is now provided inline below the radio buttons.

![image](https://github.com/carbon-design-system/carbon/assets/3360588/8dec9663-396b-4ee6-a9a8-28ef822b914f)

**Comment by tay1orjones** [comment]
After a lot of consideration and review, we're not going to land https://github.com/carbon-design-system/carbon/pull/14646 for a variety of reasons.

I know this isn't the outcome many folks wanted, but our team is committed to evaluating the root issue and providing more robust guidance around disabled elements. This work will be aimed at extending the [disabled states pattern](https://carbondesignsystem.com/patterns/disabled-states/) already present on the website. This and more is outlined in the issue I opened today:

- https://github.com/carbon-design-system/carbon-website/issues/3758

**Comment by tay1orjones** [comment]
Closing this as a lack of tooltips on disabled buttons is intended behavior
